ImageFilterEditor/ImageFilterEditor/Filtering/ImageFilterController.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  ImageFilterViewController.swift
+//  ImageFilterEditor
+//
+//  Created by Shawn Gee on 5/5/20.
+//  Copyright © 2020 Swift Student. All rights reserved.
+//
+
+import UIKit
+
+protocol ImageFilterDelegate {
+    func imageFilterDidProcessOutput(image: UIImage)
+}
+
```

Comment:
I really like the use of a navigation controller within a container view; definitely going to keep that one in mind.